src/sqlfluff/rules/L007.py

```diff
@@ -60,49 +67,149 @@ def _eval(self, context: RuleContext) -> LintResult:
-                # Prepare memory for later
-                memory["last_code"] = context.segment
-                memory["since_code"] = []
-            else:
-                # This isn't a code segment...
-                # Prepare memory for later
-                memory["since_code"].append(context.segment)
-        else:
+        is_relavent_segment = parent_stack and parent_stack[-1].is_type("expression")
+        if not is_relavent_segment:
             # Reset the memory if we're not in an expression
             memory = {"last_code": None, "since_code": []}
+            return LintResult(memory=memory)
+
+        if not context.segment.is_code:
+            # This isn't a code segment...
+            # Prepare memory for later
+            memory["since_code"].append(context.segment)
+            return LintResult(memory=memory)
+
+        # This is code, what kind?
+        if context.segment.is_type("binary_operator", "comparison_operator"):
+            # If it's an operator, then check if in "before" mode
+            if operator_new_lines == "before":
+                # If we're in "before" mode, then check if newline since last
+                # code
+                for s in memory["since_code"]:
+                    if s.name == "newline":
+                        anchor = context.segment
+                        # Had a newline - so mark this operator as a fail
+
+        elif memory["last_code"] and memory["last_code"].is_type(
+            "binary_operator", "comparison_operator"
+        ):
+            # It's not an operator, but the last code was.
+            if operator_new_lines != "before":
+                # If in "after" mode, then check to see
+                # there is a newline between us and the last operator.
+                for s in memory["since_code"]:
+                    if s.name == "newline":
+                        # Had a newline - so mark last operator as a fail
+                        anchor = memory["last_code"]
+
+        # Prepare memory for later
+        memory["last_code"] = context.segment
+        memory["since_code"] = []
 
         # Anchor is our signal as to whether there's a problem
-        if anchor:
-            return LintResult(anchor=anchor, memory=memory, description=description)
-        else:
+        if not anchor:
             return LintResult(memory=memory)
+
+        expr = cast(Tuple[BaseSegment], parent_stack[-1].segments)
+        fixes = _generate_fixes(expr, anchor, operator_new_lines)
+        return LintResult(
+            anchor=anchor,
+            memory=memory,
+            description=description,
+            fixes=fixes,
+        )
+
+
+def _generate_fixes(
+    expr: Tuple[BaseSegment],
+    anchor: BaseSegment,
+    operator_new_lines: str,
+) -> List[LintFix]:
+    """Generate Fixes.
+
+    Take operator_new_lines==after as the default case.
+    Consider "before" as a modification on this behaviour.
+
+    """
+    fixes = [LintFix.delete(anchor)]
```

Comment:
Could you add a few high-level comments to help explain the fixes being generated? "Fix" code is often a bit hard to follow.